assets/js/modules/analytics/util/data-mock.js

```diff
@@ -0,0 +1,305 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import md5 from 'md5';
+import faker from 'faker';
+import { zip, from, Observable } from 'rxjs';
+import { map, reduce, take } from 'rxjs/operators';
+
+const ANALYTICS_METRIC_TYPES = {
+	'ga:users': 'INTEGER',
+	'ga:newUsers': 'INTEGER',
+	'ga:sessions': 'INTEGER',
+	'ga:goalCompletionsAll': 'INTEGER',
+	'ga:pageviews': 'INTEGER',
+	'ga:uniquePageviews': 'INTEGER',
+	'ga:bounceRate': 'PERCENT',
+	'ga:avgSessionDuration': 'TIME',
+	'ga:adsensePageImpressions': 'INTEGER',
+	'ga:adsenseCTR': 'PERCENT',
+	'ga:adsenseRevenue': 'CURRENCY',
+	'ga:adsenseECPM': 'CURRENCY',
+};
+
+const ANALYTICS_DIMENSION_OPTIONS = {
+	'ga:channelGrouping': [
+		'Organic Search',
+		'Referral',
+		'Direct',
+		'(other)',
+	],
+	'ga:country': [
+		'United States',
+		'United Kingdom',
+		'India',
+		'(not set)',
+		'France',
+		'Ukraine',
+		'Italy',
+		'Mexico',
+	],
+	'ga:deviceCategory': [
+		'desktop',
+		'tablet',
+		'mobile',
+	],
+	'ga:pageTitle': ( i ) => i <= 12 ? `Test Post ${ i }` : false,
+	'ga:pagePath': ( i ) => i <= 12 ? `/test-post-${ i }/` : false,
+};
+
+/**
+ * Gets metric key.
+ *
+ * @since n.e.x.t
+ *
+ * @param {string|Object} metric Metric name or object.
+ * @return {string} Metric key.
+ */
+function getMetricKey( metric ) {
+	return metric?.expression || metric.toString();
+}
+
+/**
+ * Gets metric type.
+ *
+ * @since n.e.x.t
+ *
+ * @param {string|Object} metric Metric name or object.
+ * @return {string} Type of the metric.
+ */
+function getMetricType( metric ) {
+	return ANALYTICS_METRIC_TYPES[ getMetricKey( metric ) ];
+}
+
+/**
+ * Generates and returns metric values.
+ *
+ * @since n.e.x.t
+ *
+ * @param {string} type  Metric type.
+ * @param {number} count Maximum number of values to generate.
+ * @return {Array.<Object>} Array of metric values.
+ */
+function generateMetricValues( type, count ) {
+	const metrics = [];
+
+	for ( let i = 0; i < count; i++ ) {
+		const values = [];
+
+		switch ( type ) {
+			case 'INTEGER':
+				values.push( faker.random.number( { min: 0, max: 100 } ).toString() );
+				break;
+			case 'PERCENT':
+				values.push( faker.random.float( { min: 0, max: 100 } ).toString() );
+				break;
+			case 'TIME':
+				values.push( faker.random.number( { min: 0, max: 3600 } ).toString() ); // 1 hour max.
+				break;
+			case 'CURRENCY':
+				values.push( faker.random.float( { min: 0, max: 10000 } ).toString() ); // $10k max.
+				break;
```

Comment:
I'm not sure about this approach of using random numbers. While overall this is great, it would cause problems as soon as we start running VRT on some of these components. I think it would be safer to use a list of static realistic values and combine them somehow (similar to the PHP-based solution in the Site Kit tester plugin), so that the mock responses would be stable.

Reply:
@felixarntz I think this is still the case. The numbers aren't entirely random; they're randomish based on the seed faker is given which is what makes them consistently reproducible. You can see what I mean here:

```
> var faker = require('faker')
undefined
> faker.seed(123)
undefined
> faker.random.number()
69646
> faker.random.number()
71295
> faker.random.number()
28613
> faker.seed(123)
undefined
> faker.random.number()
69646
> faker.random.number()
71295
> faker.random.number()
28613
```

Reply:
Yep, we use a seed value generated from the options URL. It guarantees that generated values will be the same everywhere. See here: https://github.com/google/site-kit-wp/blob/dcb62f455597783e1573033e710f9723f41da055/assets/js/modules/analytics/util/data-mock.js#L177-L187